ibis/backends/bigquery/__init__.py

```diff
@@ -232,7 +232,7 @@ def dataset_id(self):
         return self.dataset
 
     def table(self, name: str, database: str | None = None) -> ir.TableExpr:
-        if database is None:
+        if database is None and self._fully_qualified_name(name, database) != name:
             database = f"{self.data_project}.{self.current_database}"
         t = super().table(name, database=database)
         table_id = self._fully_qualified_name(name, database)
```

Comment:
```suggestion
        table_id = self._fully_qualified_name(name, database)
        t = super().table(table_id)
```

Reply:
Sure thing. I am working on it.

Thank you for the comment.